### PR TITLE
Add and use connectivity package for states

### PR DESCRIPTION
**Pull request by menghanl**

### CODE REVIEW

File: clientconn.go [hunk 1]
```diff
@@ -587,9 +554,9 @@ type ClientConn struct {
 	mkp keepalive.ClientParameters
 }
 
-// WaitForStateChange waits until the ConnectivityState of ClientConn changes from sourceState or
+// WaitForStateChange waits until the connectivity.State of ClientConn changes from sourceState or
 // ctx expires. A true value is returned in former case and false in latter.
```
Comment (dfawley) [inline clientconn.go]: Yeesh, we didn't write the "experimental" disclaimer here.  Can you add it?

File: clientconn.go [hunk 2]
```diff
@@ -602,8 +569,8 @@ func (cc *ClientConn) WaitForStateChange(ctx context.Context, sourceState Connec
 	}
 }
 
-// GetState returns the ConnectivityState of ClientConn.
-func (cc *ClientConn) GetState() ConnectivityState {
+// GetState returns the connectivity.State of ClientConn.
```
Comment (dfawley) [inline clientconn.go]: Same.

**Comment by dfawley** [comment]
@dzbarsky FYI, we will be changing the connectivity API slightly in this PR (to support upcoming work revamping the resolver and balancer logic), in case you are using it already.

File: clientconn.go (hunk 1 above)
Comment (menghanl) [inline clientconn.go]: Done.

File: clientconn.go (hunk 2 above)
Comment (menghanl) [inline clientconn.go]: Done

**Comment by dzbarsky** [comment]
@dfawley thanks for the warning, we're already using this since we were blocked on it for a while. It looks like this just moved it to a separate package so that shouldn't be a problem for us. Please let us know if there are any actual breaking changes, thanks!

**Pull request merged**